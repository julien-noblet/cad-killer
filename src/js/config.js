/**
 * @flow
 *
 * @format
 */

export const CENTER = [46.495, 2.201];
export const API_URL = "//api-adresse.data.gouv.fr/search/?";
export const REVERSE_URL = "//api-adresse.data.gouv.fr/reverse/?";
export const SHORT_CITY_NAMES = [
  "y",
  "ay",
  "bu",
  "by",
  "eu",
  "fa",
  "gy",
  "oo",
  "oz",
  "py",
  "ri",
  "ry",
  "sy",
  "ur",
  "us",
  "uz",
];
export const ATTRIBUTIONS =
  "&copy; <a href='http://www.openstreetmap.org/copyright'>Contributeurs de OpenStreetMap</a> | <a href='https://www.data.gouv.fr/fr/datasets/base-d-adresses-nationale-ouverte-bano/'>Adresses BAN</a> sous licence ODbL";
export const IGN_KEY = "zh4iqh58bg7a17hqzo16nqfw";
export const IGN_LAYER = "GEOGRAPHICALGRIDSYSTEMS.MAPS";
export const IGN_LAYER_LITE = "GEOGRAPHICALGRIDSYSTEMS.PLANIGN";
export const MY_POUCHDB =
  "//sidercomourellydiforteep:16f21a8ad23f32de280b36fc338a38a24c5d49b2@couchdb-112956.smileupps.com/stats";
export const LOCAL_POUCHDB = "cad-killer";
export const OSM_CREDITENTIALS = "Q0FELUtJTExFUjpkdHl2dWRlbnQ=";
export const NOTE_API = "//api.openstreetmap.org/api/0.6/notes";
