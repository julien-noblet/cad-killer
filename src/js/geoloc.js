/**
 * global L, map
 *
 * @format
 */

// GÃ©oloc
function showPosition(position) {
  const icone = L.DomUtil.get(document.getElementById("geoloc_icon"));
  if (Window.map !== null) {
    Window.map.setView(
      [position.coords.latitude, position.coords.longitude],
      16
    );
    icone.className = "zmdi zmdi-2x zmdi-gps-dot";
  }
}

/* eslint-disable no-unused-vars */
function getLocation() {
  const icone = document.getElementById("geoloc_icon");
  if (navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(showPosition);
    icone.className = "zmdi zmdi-2x zmdi-gps";
  }
}
/* eslint-enable no-unused-vars */

const GeoLoc = L.Control.extend({
  options: {
    position: "topright",
  },
  onAdd: () => {
    const container = L.DomUtil.create("div", "leaflet-control-geoloc");
    container.innerHTML =
      '<span onClick="getLocation();" id="geoloc" class="geoloc"><i class="zmdi zmdi-2x zmdi-gps-off" id="geoloc_icon"></i></span>';
    // ... initialize other DOM elements, add listeners, etc.
    return container;
  },
});

window.map.addControl(new GeoLoc());
