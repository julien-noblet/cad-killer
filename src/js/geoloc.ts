/**
 * global L, map
 *
 * @format
 */

import * as L from "leaflet";
import { mymap as map } from "./map";

// GÃ©oloc
function showPosition(position: { coords: { latitude: any; longitude: any; }; }) {
  const icone = L.DomUtil.get(document.getElementById("geoloc_icon"));
  if (map !== null) {
    map.setView(
      [position.coords.latitude, position.coords.longitude],
      16
    );
    icone.className = "zmdi zmdi-2x zmdi-gps-dot";
  }
}

/* eslint-disable no-unused-vars */
function getLocation() {
  const icone = document.getElementById("geoloc_icon");
  if (navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(showPosition);
    icone.className = "zmdi zmdi-2x zmdi-gps";
  }
}
/* eslint-enable no-unused-vars */

const GeoLoc = L.Control.extend({
  options: {
    position: "topright",
  },
  onAdd: () => {
    const container = L.DomUtil.create("div", "leaflet-control-geoloc");
    container.innerHTML =
      '<span onClick="getLocation();" id="geoloc" class="geoloc"><i class="zmdi zmdi-2x zmdi-gps-off" id="geoloc_icon"></i></span>';
    // ... initialize other DOM elements, add listeners, etc.
    return container;
  },
});

map.addControl(new GeoLoc());
