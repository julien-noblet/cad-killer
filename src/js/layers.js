/**
 * @flow
 *
 * @format
 */

import L from "leaflet";
import { IGN_KEY, IGN_LAYER } from "./config";

// Layers
export const layerOSMfr: any = L.tileLayer(
  "//{s}.tile.openstreetmap.fr/osmfr/{z}/{x}/{y}.png",
  {
    maxZoom: 20,
    attribution:
      'Fond de plan &copy; <a href="https://openstreetmap.fr/">OpenStreetMap France</a>',
  }
);

export const layerOSM: any = L.tileLayer(
  "//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
  {
    maxZoom: 19,
    attribution:
      'Fond de plan &copy; <a href="https://openstreetmap.org/">OpenStreetMap</a>',
  }
);

export const layerCadastre: any = L.tileLayer(
  "http://tms.cadastre.openstreetmap.fr/*/tout/{z}/{x}/{y}.png",
  {
    maxZoom: 22,
    minZoom: 16,
    attribution: "&copy; Cadastre",
  }
);

export const overlayCadastre: any = L.tileLayer(
  "http://tms.cadastre.openstreetmap.fr/*/transp/{z}/{x}/{y}.png",
  {
    maxZoom: 22,
    minZoom: 16,
    attribution: "&copy; Cadastre",
  }
);

export const layerEsriWorldImagery: any = L.tileLayer(
  "//server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}",
  {
    attribution:
      "&copy; Esri &mdash; Source: Esri, i-cubed, USDA, USGS, AEX, GeoEye, Getmapping, Aerogrid, IGN, IGP, UPR-EGP, and the GIS User Community",
  }
);


export const layerEsriWorldStreetMap: any = L.tileLayer(
  "//server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}",
  {
    attribution:
    'Tiles &copy; Esri &mdash; ' +
    'Source: Esri, DeLorme, NAVTEQ, USGS, Intermap, iPC, NRCAN, Esri Japan, METI, Esri China (Hong Kong), Esri (Thailand), TomTom, 2012'
  }
);

export const layerIGN: any = L.tileLayer(
  `//data.geopf.fr/wmts?SERVICE=WMTS&REQUEST=GetTile&VERSION=1.0.0&LAYER=${IGN_LAYER}&STYLE=normal&FORMAT=image/jpeg&TILEMATRIXSET=PM&TILEMATRIX={z}&TILEROW={y}&TILECOL={x}`,
 // `//wxs.ign.fr/${IGN_KEY}/geoportail/wmts?SERVICE=WMTS&REQUEST=GetTile&VERSION=1.0.0&LAYER=${IGN_LAYER}&STYLE=normal&TILEMATRIXSET=PM&TILEMATRIX={z}&TILEROW={y}&TILECOL={x}&FORMAT=image%2Fjpeg`,
  {
    maxZoom: 18,
    minZoom: 6,
    attribution: 'IGN-F/Géoportail',
    tileSize : 256 // les tuiles du Géooportail font 256x256px
  }
);

export const overlayBAN: any = L.tileLayer(
  "//{s}.layers.openstreetmap.fr/bano/{z}/{x}/{y}.png",
  {
    maxZoom: 20,
    attribution: "Surcouche: &copy; BAN(O)",
  }
);

export const baseMaps = {
  "OpenStreetMap France": layerOSMfr,
  OpenStreetMap: layerOSM,
  // "Carte IGN": layerIGN, // je n'arrive pas a renouveller la clé IGN pour le moment
  Cadastre: layerCadastre,
  Esri: layerEsriWorldImagery,
  "World Street Map" : layerEsriWorldStreetMap,
};

export const overlayMaps = {
  Cadastre: overlayCadastre,
  "BAN(O)": overlayBAN,
};
