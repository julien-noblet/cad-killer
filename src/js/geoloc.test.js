/** @format */

import L from "leaflet";
import { ATTRIBUTIONS, CENTER } from "./config";
import { overlayMaps, baseMaps, layerOSMfr } from "./layers";

const div = global.document.createElement("div");
div.setAttribute("id", "map");
global.document.body.appendChild(div);
window.map = L.map("map", {
  attributionControl: false
});
map = window.map;
describe("test Geoloc", () => {
  test("geoloc_icon exist", () => {
    require("./geoloc");
    expect(document.getElementById("geoloc_icon")).toExist;
  });
  test("geoloc_icon is correct", () => {
    expect(document.getElementById("geoloc_icon")).toMatchSnapshot();
  });
});
