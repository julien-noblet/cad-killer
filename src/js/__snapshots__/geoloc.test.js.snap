// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Geoloc geoloc_icon is correct 1`] = `
<i
  class="zmdi zmdi-2x zmdi-gps-off"
  id="geoloc_icon"
/>
`;
